### ISSUE TITLE
Bind picker view data source

### ISSUE THREAD

**Issue by jonathanfoster** [DeclarativeHub/Bond#577]
Is there a way to bind UIPickerView data source similar to UITableView? For instance, if using an enum as the data source:

```
var genderOptions = ObservableArray<GenderType>(GenderType.allCases)

genderOptions.bind(to: self.genderPicker) { (dataSource, row, component, pickerView) -> String? in
    return dataSource.array[row].rawValue
 }
```

Or using a string array:

```
var genderOptions = ObservableArray<String>(["Male", "Female"])

genderOptions.bind(to: self.genderPicker)
```

**Comment by jonathanfoster** [comment]
I threw together a potential implementation based on UITableView binding extensions. If this is headed in the right direction, then I could submit a PR and work feedback from there. 

https://github.com/jonathanfoster/Bond/blob/pickerview-bind/Sources/Bond/UIKit/UIPickerView.swift

**Comment by npvisual** [comment]
@jonathanfoster : nice implementation. I started using this in my project since I wanted to use bind(to:) on UIPickerView.

However, I had to change the Element conformance to 'SectionedDataSourceChangesetConvertible' to be able to bind an ObservableArray to UIPickerView (and a few more lines).

```
public class PickerViewBinder<DataSource: SectionedDataSourceChangesetConvertible> {
    public var createTitle: ((DataSource, Int, Int, UIPickerView) -> String?)?
...

extension SignalProtocol where Element: SectionedDataSourceChangesetConvertible, Error == NoError {
    public typealias DataSource = Element
...

            map: { (dataSource: DataSource?, pickerView: UIPickerView) -> Int in
                return dataSource?.asSectionedDataSourceChangeset.collection.numberOfSections ?? 0
...

            map: { (dataSource: DataSource?, pickerView: UIPickerView, component: Int) -> Int in
                return dataSource?.asSectionedDataSourceChangeset.collection.numberOfItems(inSection: component) ?? 0
```

Not sure if that's the right thing to do or not, but once that was done, things worked really well.

**Comment by jonathanfoster** [comment]
@npvisual: Thanks for the update. I ended up going a different direction for the use case that prompted this issue so I abandoned the effort.

I just ran into the same use case again so I'm going to revisit this issue and update the fork. I'll try to get whatever I come up with upstream. I'm upping the difficulty level this time, I need a 2D array bound for height (feet and inches options). I'll start with the simple 1D array use case first though.
